src/cascadia/TerminalConnection/AzureConnection.cpp

```diff
@@ -889,99 +921,91 @@ namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
+                { winrt::hstring{ L"grant_type" }, winrt::hstring{ L"refresh_token" } },
+                { winrt::hstring{ L"client_id" }, winrt::hstring{ AzureClientID } },
+                { winrt::hstring{ L"resource" }, winrt::hstring{ _wantedResource } },
+                { winrt::hstring{ L"refresh_token" }, winrt::hstring{ _refreshToken } },
+            }
+        };
 
         // Send the request and return the response as a json value
-        auto refreshResponse{ _SendRequestReturningJson(refreshClient, refreshRequest) };
-        _accessToken = refreshResponse.at(L"access_token").as_string();
-        _refreshToken = refreshResponse.at(L"refresh_token").as_string();
-        _expiry = std::stoi(refreshResponse.at(L"expires_on").as_string());
+        auto refreshResponse{ _SendRequestReturningJson(uri, content) };
+        _setAccessToken(refreshResponse.GetNamedString(L"access_token"));
+        _refreshToken = refreshResponse.GetNamedString(L"refresh_token");
+        _expiry = std::stoi(winrt::to_string(refreshResponse.GetNamedString(L"expires_on")));
     }
 
     // Method description:
     // - helper function to get the user's cloud shell settings
     // Return value:
     // - the user's cloud shell settings
-    json::value AzureConnection::_GetCloudShellUserSettings()
+    WDJ::JsonObject AzureConnection::_GetCloudShellUserSettings()
     {
-        // Initialize client
-        http_client settingsClient(_resourceUri);
-
-        // Initialize request
-        http_request settingsRequest(L"GET");
-        settingsRequest.set_request_uri(L"providers/Microsoft.Portal/userSettings/cloudconsole?api-version=2018-10-01");
-
-        return _SendAuthenticatedRequestReturningJson(settingsClient, settingsRequest);
+        auto uri{ fmt::format(L"{}providers/Microsoft.Portal/userSettings/cloudconsole?api-version=2020-04-01-preview", _resourceUri) };
+        return _SendRequestReturningJson(uri, nullptr);
     }
 
     // Method description:
     // - helper function to request for a cloud shell
     // Return value:
     // - the uri for the cloud shell
-    utility::string_t AzureConnection::_GetCloudShell()
+    winrt::hstring AzureConnection::_GetCloudShell()
     {
-        // Initialize client
-        http_client cloudShellClient(_resourceUri);
+        auto uri{ fmt::format(L"{}providers/Microsoft.Portal/consoles/default?api-version=2020-04-01-preview", _resourceUri) };
 
-        // Initialize request
-        http_request shellRequest(L"PUT");
-        shellRequest.set_request_uri(L"providers/Microsoft.Portal/consoles/default?api-version=2018-10-01");
-        // { "properties": { "osType": "linux" } }
-        auto body = json::value::object({ { U("properties"), json::value::object({ { U("osType"), json::value::string(U("linux")) } }) } });
-        shellRequest.set_body(body);
+        WWH::HttpStringContent content{
+            LR"-({"properties": {"osType": "linux"}})-",

```

Comment:
I know. We were constructing this like a real JSON object. It just feels... wasteful.